WordPress/gutenberg
# Docs: Organize and update block theme docs

## Description

A follow up PR to #31065

Updates and organizes the block theme documentation. An overview document was missing from the index, plus naming and consistency around using "block theme"

The previous themes/ directory was removed it contained an extra layer to click through to get to the single theme-support file, plus having two sections about themes is confusing.

- Move unlinked overview document to `how-to-guides/themes/block-theme-overview.md`
- Move previous `block-theme/README.md` to `themes/create-block-theme.md`
- Update titles to simplify in sidebar and reflect new names
- Update references of "block-based theme" to "block theme"
- Search through repo and update incoming links

## How has this been tested?

Documentation update.

- Confirm new organization makes sense
- Confirm titles and minor text changes are accurate
- Confirm links work

You can browse the changes [on the branch rendered here](https://github.com/WordPress/gutenberg/tree/docs/block-theme-dir/docs/how-to-guides).


## Types of changes

- Documentation
- Updates toc.json and manifest.json
- Meta Trac Ticket for redirects at: https://meta.trac.wordpress.org/ticket/5714
